src/components/dialogs/parameters/util/make-component-utils.js

```diff
@@ -6,110 +6,42 @@
-export const makeRenderSwitchWithLabel =
-    () => (defParam, key, params, setter, value) => {
-        return (
-            <SwitchWithLabel
-                value={value}
-                label={defParam.description}
-                callback={(ev) =>
-                    setter({ ...params, [key]: ev.target.checked })
-                }
-            />
-        );
-    };
-export const makeRenderDropDown =
-    () => (defParam, key, params, setter, value) => {
-        return (
-            <DropDown
-                value={value}
-                label={defParam.description}
-                values={defParam.values}
-                callback={(ev) => setter({ ...params, [key]: ev.target.value })}
-            />
-        );
-    };
-
-export const makeRenderIntegerField =
-    () => (defParam, key, params, setter, value) => {
-        return (
-            <IntegerInput
-                value={value}
-                label={defParam.description}
-                callback={({ value }) => setter({ ...params, [key]: value })}
-            />
-        );
-    };
-export const makeRenderFloatField =
-    () => (defParam, key, params, setter, value) => {
-        return (
-            <FloatInput
-                value={value}
-                label={defParam.description}
-                callback={({ value }) => setter({ ...params, [key]: value })}
-            />
-        );
-    };
-
-export const makeRenderTextField =
-    () => (defParam, key, params, setter, value) => {
-        return (
-            <TextInput
-                value={value}
-                label={defParam.description}
-                callback={({ value }) => setter({ ...params, [key]: value })}
-            />
-        );
-    };
-
-// --- define data types --- //
-export const TYPES = {
-    enum: 'Enum',
-    bool: 'Bool',
-    integer: 'Integer',
-    float: 'Float',
-    text: 'Text',
-};
-
-// --- define default render for each data types
-const DEFAULT_RENDER = {
-    [TYPES.bool]: makeRenderSwitchWithLabel(),
-    [TYPES.enum]: makeRenderDropDown(),
-    [TYPES.integer]: makeRenderIntegerField(),
-    [TYPES.float]: makeRenderFloatField(),
-    [TYPES.text]: makeRenderTextField(),
-};
-
-export const makeComponentsFor = (defParams, params, setter) => {
+export const makeComponents = (defParams) => {
     return Object.keys(defParams).map((key) => (
         <Grid container spacing={1} paddingTop={1} key={key}>
-            {makeComponentFor(defParams[key], key, params, setter)}
+            {makeComponent(defParams[key], key)}
             <LineSeparator />
         </Grid>
     ));
 };
 
+export const makeComponent = (defParam, key) => {
+    const render = defParam.render;
+
+    return (
+        <>
+            <Grid item xs={8} sx={styles.parameterName}>
+                <FormattedMessage id={defParam.label} />
+            </Grid>
+            <Grid item container xs={4} sx={styles.controlItem}>
+                {render(defParam, key)}
```

Comment:
I use a lazy technique which enable to refer to a value lately at the rendering time instead of declaration time. By this way, we can configure more flexibly the render..